File: backend/app/models/project.py
from datetime import datetime


def new_project(
    title: str,
    client_id: str,
    description: str = "",
    categories: list[str] | None = None,
) -> dict:
    return {
        "title": title,
        "description": description,
        "client_id": client_id,
        "status": "active",
        "cover_image_key": None,
        "categories": categories or [],
        "share_link_token": None,
        "share_link_expires_at": None,
        "project_expires_at": None,
        "created_at": datetime.utcnow(),
        "updated_at": datetime.utcnow(),
    }

File: src/models/project.ts
export function newProject(title: string, clientId: string, description = '', categories: string[] = []) {
  const now = new Date();
  return {
    title,
    description,
    client_id: clientId,
    status: 'active',
    cover_image_key: null as string | null,
    categories,
    share_link_token: null as string | null,
    share_link_expires_at: null as Date | null,
    project_expires_at: null as Date | null,
    gallery_consent: {
      status: null as 'agree' | 'withdrawn' | null,
      signed_at: null as Date | null,
      signed_ip: null as string | null,
      withdrawn_at: null as Date | null,
    },
    created_at: now,
    updated_at: now,
  };
}
